engine/gamesys/src/gamesys/scripts/script_physics.cpp

```diff
@@ -1453,6 +1552,42 @@ namespace dmGameSystem
         return 0;
     }
 
+    static int Physics_SetListener(lua_State* L)
+    {
+        DM_LUA_STACK_CHECK(L, 0);
+        int top = lua_gettop(L);
+        int functionref = 0;
+        if (top >= 1) // completed cb
+        {
```

Comment:
If I reset the listender to `nil`, who decref's the callback value?
What is the intended lifespan here?

Reply:
it's a good point, do we need an ability set it to nil?

Reply:
Now it uses callbacks